mmf/trainers/lightning_trainer.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+import logging
+import math
+
+import omegaconf
+import torch
+from mmf.common import typings as mmf_typings
+from mmf.common.registry import registry
+from mmf.datasets.lightning_datamodule import LightningDataModule
+from mmf.modules.metrics import Metrics
+from mmf.trainers.base_trainer import BaseTrainer
+from mmf.trainers.lightning_core.loop_callback import LightningLoopCallback
+from mmf.utils.build import build_model
+from mmf.utils.general import get_max_updates, print_model_parameters
+from pytorch_lightning import Trainer
+
+
+logger = logging.getLogger(__name__)
+
+
+@registry.register_trainer("lightning")
+class LightningTrainer(BaseTrainer):
+    def __init__(self, config: mmf_typings.DictConfig):
+        super().__init__(config)
+        self.trainer = None
+
+    def load(self):
+        super().load()
+        self._calculate_max_updates()
+        self._calculate_gradient_clip_val()
+        self._load_trainer()
+
+    def _load_trainer(self):
+        self.trainer = Trainer(
+            logger=False,
+            gpus=self._gpus,
+            num_nodes=self._num_nodes,
+            callbacks=self._callbacks,
+            precision=16 if self.training_config.fp16 else 32,
+            deterministic=self._deterministic,
+            benchmark=self._benchmark,
+            max_steps=self._max_updates,
+            max_epochs=self._max_epochs,
+            gradient_clip_val=self._gradient_clip_val,
+            num_sanity_val_steps=0,
+            automatic_optimization=True,
+            checkpoint_callback=False,
+            accumulate_grad_batches=self.training_config.update_frequency,
+            val_check_interval=self.training_config.evaluation_interval,
+            log_every_n_steps=self.training_config.log_interval,
+        )
```

Comment:
Here is the problem I was talking about. Now, users need to learn extra stuff such as what do lightning parameters correspond to in MMF which is unnecessary. We are losing one of the benefits of moving to lightning as these parameters would be better documented in Lightning. 

Also, if something else is added here tomorrow, we will need to again update our training creating an unnecessary dependency loop.

Reply:
Agree. These should be passed as parameters to the lightning trainer config directly.